biomage/experiment/download.py

```diff
@@ -0,0 +1,338 @@
+import json
+import os
+from enum import Enum
+from fileinput import filename
+from pathlib import Path
+
+import boto3
+import click
+from botocore.exceptions import ClientError
+
+from ..rds.run import run_rds_command
+from ..utils.constants import (CELLSETS_BUCKET, PROCESSED_RDS_BUCKET,
+                               RAW_RDS_BUCKET, SAMPLES_BUCKET, SAMPLES_TABLE)
+from .utils import set_modified_date
+
+output_path = "."
+
+
+s3 = boto3.resource("s3")
+SAMPLES = "samples"
+RAW_RDS = "raw_rds"
+PROCESSED_RDS = "qc_rds"
+CELLSETS = "cellsets"
+
+
+def _download_file(bucket, s3_path, file_path):
+    """
+    Utility to download file
+    """
+    file_path.parent.mkdir(parents=True, exist_ok=True)
+
+    s3_obj = s3.Object(bucket, s3_path)
+    s3_obj.download_file(str(file_path))
+    set_modified_date(file_location=file_path, date=s3_obj.last_modified)
+
+
+def _create_sample_mapping(samples_list, output_path):
+    """
+    Create a mapping of sample names to sample ids.
+    """
+
+    MAPPING_FILE_NAME = "sample_mapping.json"
+
+    samples_file = output_path / MAPPING_FILE_NAME
+
+    sample_mapping = {}
+
+    for sample_name, samples in samples_list.items():
+        sample_mapping[sample_name] = samples[0]["sample_id"]
+
+    samples_file.write_text(json.dumps(sample_mapping))
+
+    print(f"Sample name-id map downloaded to: {str(samples_file)}.\n")
+
+
+def _download_samples_v1(experiment_id, input_env, output_path, name_sample_using_id):
```

Comment:
Why is this function necessary? We are very soon not going to use v1 and bioinformaticians don't have trouble downloading from v1

Reply:
Some experiments are not yet available in v2, that's why I created support for this. But you have a point - we don't need v1 anymore. I'll log a message telling the user to downlod the files directly from S3 if it's not available in v2.

Reply:
sounds good, Agi